engine/src/main/java/org/terasology/logic/inventory/ItemComponent.java

```diff
@@ -49,7 +49,7 @@
      * How many of said item are there in this stack
      */
     @Replicate(FieldReplicateType.SERVER_TO_CLIENT)
-    public byte stackCount = 1;
+    public int stackCount = 1;
```

Comment:
You make stackCount and int but maxStackCount remains a byte. It's always good when a PR changes just one thing and not multiple. So I would suggest you revert this change.